crates/router/src/routes/recon.rs

```diff
@@ -0,0 +1,249 @@
+    let flow = Flow::ReconTokenRequest;
+    Box::pin(api::server_wrap(
+        flow,
+        state,
+        &req,
+        (),
+        |state, user: ReconUser, _| generate_recon_token(state, user),
+        &auth::ReconJWT,
+        api_locking::LockAction::NotApplicable,
+    ))
+    .await
+}
+
+pub async fn send_recon_request(
+    state: AppState,
+    user: UserFromToken,
+) -> RouterResponse<recon_api::ReconStatusResponse> {
+    let db = &*state.store;
+    let user_from_db = db
+        .find_user_by_id(&user.user_id)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+    let merchant_id = db
+        .find_user_role_by_user_id(&user.user_id)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?
+        .merchant_id;
+    let key_store = db
+        .get_merchant_key_store_by_merchant_id(
+            merchant_id.as_str(),
+            &db.get_master_key().to_vec().into(),
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+    let merchant_account = db
+        .find_merchant_account_by_merchant_id(merchant_id.as_str(), &key_store)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+
+    let email_contents = email_types::ProFeatureRequest {
+        feature_name: "RECONCILIATION & SETTLEMENT".to_string(),
+        merchant_id: merchant_id.clone(),
+        user_name: UserName::new(user_from_db.name)
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failed to form username")?,
+        recipient_email: UserEmail::from_pii_email(user_from_db.email.clone())
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failed to convert to UserEmail from pii::Email")?,
+        settings: state.conf.clone(),
+        subject: format!(
+            "Dashboard Pro Feature Request by {}",
+            user_from_db.email.expose().peek()
+        ),
+    };
+
+    let is_email_sent = state
+        .email_client
+        .compose_and_send_email(
+            Box::new(email_contents),
+            state.conf.proxy.https_url.as_ref(),
+        )
+        .await
+        .change_context(UserErrors::InternalServerError)
+        .map_err(|error| {
+            logger::error!(?error);
+            error
+        })
+        .is_ok();
+
+    if is_email_sent {
+        let updated_merchant_account = storage::MerchantAccountUpdate::ReconUpdate {
+            recon_status: enums::ReconStatus::Requested,
+        };
+
+        let response = db
+            .update_merchant(merchant_account, updated_merchant_account, &key_store)
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable_lazy(|| format!("Failed while updating merchant: {merchant_id}"))?;
```

Comment:
```suggestion
            .attach_printable_lazy(|| format!("Failed while updating recon status for merchant: {merchant_id}"))?;
```

Please make the same change elsewhere in this file.